lib/Runtime/PlatformAgnostic/Platform/Common/Intl.cpp

```diff
@@ -0,0 +1,485 @@
+        AutoPtr<icu::NumberFormat> guard(nf); // ICU requires that the caller explicitly deletes pointers allocated by ICU (otherwise will leak)
+
+        if (U_FAILURE(error))
+        {
+#ifdef INTL_ICU_DEBUG
+            if (error == UErrorCode::U_MISSING_RESOURCE_ERROR)
+            {
+                Output::Print(_u("EntryIntl_CurrencyDigits > icu::NumberFormat::createCurrencyInstance(error) > U_MISSING_RESOURCE_ERROR (%d)\n"), error);
+            }
+            else
+            {
+                Output::Print(_u("EntryIntl_CurrencyDigits > icu::NumberFormat::createCurrencyInstance(error) > UErrorCode (%d)\n"), error);
+            }
+#endif
+            goto LReturn;
+        }
+
+        nf->setCurrency(uCurrencyCode, error);
+        if (U_FAILURE(error))
+        {
+#ifdef INTL_ICU_DEBUG
+            if (error == UErrorCode::U_MISSING_RESOURCE_ERROR)
+            {
+                Output::Print(_u("EntryIntl_CurrencyDigits > nf->setCurrency(uCurrencyCode (%s), error) > U_MISSING_RESOURCE_ERROR (%d)\n"), currencyCode, error);
+            }
+            else
+            {
+                Output::Print(_u("EntryIntl_CurrencyDigits > nf->setCurrency(uCurrencyCode (%s), error) > UErrorCode (%d)\n"), currencyCode, error);
+            }
+#endif
+            goto LReturn;
+        }
+
+        minFracDigits = nf->getMinimumFractionDigits();
+
+#ifdef INTL_ICU_DEBUG
+        Output::Print(_u("EntryIntl_CurrencyDigits > nf->getMinimumFractionDigits() successful > returned (%d)\n"), minFracDigits);
+#endif
+
+    LReturn:
+#ifdef INTL_ICU_DEBUG
+        Output::Print(_u("EntryIntl_CurrencyDigits > returning (%d)\n"), minFracDigits);
+#endif
+
+        return minFracDigits;
+    }
+
+    template <typename Func>
+    HRESULT CreateFormatter(Func function, _In_z_ const char16 *languageTag, _In_ const charcount_t cch, _Out_ IPlatformAgnosticResource **resource)
+    {
+        UErrorCode error = UErrorCode::U_ZERO_ERROR;
+
+        // Allocate memory for the UTF8 output buffer. Need 3 bytes for each (code point + null) to satisfy SAL.
+        const size_t inputLangTagUtf8SizeAllocated = AllocSizeMath::Mul(AllocSizeMath::Add(cch, 1), 3);
+        unsigned char *inputLangTagUtf8 = new unsigned char[inputLangTagUtf8SizeAllocated];
+        StringBufferAutoPtr<unsigned char> guard(inputLangTagUtf8);
+
+        if (!inputLangTagUtf8)
+        {
+            AssertOrFailFastMsg(false, "OOM: failed to allocate inputLangTagUtf8.");
+            return E_OUTOFMEMORY;
+        }
+
+        // explicitly zero the allocated array
+        for (size_t index = 0; index < inputLangTagUtf8SizeAllocated; ++index)
+        {
+            inputLangTagUtf8[index] = 0;
+        }
+
+        utf8::EncodeIntoAndNullTerminate(inputLangTagUtf8, languageTag, cch);
+
+        char *localeName = reinterpret_cast<char *>(inputLangTagUtf8);
+        auto locale = icu::Locale::createFromName(localeName);
```

Comment:
Just a nitpick but I'm not a big fan of `auto`.

Reply:
Some of this code was more readable with `auto` -- this one is debatable. It's an `icu::Locale` and the type is on the RHS so I thought maybe. But yeah, I don't like this one either. I'll change it.